src/Entity/Rule.php

```diff
@@ -0,0 +1,345 @@
+     *
+     * @param string $label
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Rule
+     */
+    public function setLabel($label)
+    {
+        if (!is_string($label)) {
+            throw new LiftSdkException('Argument must be an instance of string.');
+        }
+        $this['label'] = $label;
+
+        return $this;
+    }
+
+    /**
+     * Gets the 'label' parameter.
+     *
+     * @return string
+     */
+    public function getLabel()
+    {
+        return $this->getEntityValue('label', '');
+    }
+
+    /**
+     * Sets the 'description' parameter.
+     *
+     * @param string $description
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Rule
+     */
+    public function setDescription($description)
+    {
+        if (!is_string($description)) {
+            throw new LiftSdkException('Argument must be an instance of string.');
+        }
+        $this['description'] = $description;
+
+        return $this;
+    }
+
+    /**
+     * Gets the 'description' parameter.
+     *
+     * @return string The Description of the Rule
+     */
+    public function getDescription()
+    {
+        return $this->getEntityValue('description', '');
+    }
+
+    /**
+     * Sets the 'slot_id' parameter.
+     *
+     * @param string $slotId
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Rule
+     */
+    public function setSlotId($slotId)
+    {
+        if (!is_string($slotId)) {
+            throw new LiftSdkException('Argument must be an instance of string.');
+        }
+        $this['slot_id'] = $slotId;
+
+        return $this;
+    }
+
+    /**
+     * Gets the 'description' parameter.
+     *
+     * @return string The Description of the Rule
+     */
+    public function getSlotId()
+    {
+        return $this->getEntityValue('slot_id', '');
+    }
+
+    /**
+     * Sets the 'weight' parameter.
+     *
+     * @param int $weight
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Rule
+     */
+    public function setWeight($weight)
+    {
+        if (!is_integer($weight)) {
+            throw new LiftSdkException('Argument must be an instance of integer.');
+        }
+        $this['weight'] = $weight;
+
+        return $this;
+    }
+
+    /**
+     * Gets the 'description' parameter.
+     *
+     * @return int The Description of the Rule
+     */
+    public function getWeight()
+    {
+        return $this->getEntityValue('weight', 0);
+    }
+
+    /**
+     * Sets the 'content' parameter.
+     *
+     * @param Content[] $contentList
+     *
+     * @throws \Acquia\LiftClient\Exception\LiftSdkException
+     *
+     * @return \Acquia\LiftClient\Entity\Rule
+     */
+    public function setContent(array $contentList)
```

Comment:
Hmm, this a tricky one. I wanted to stay as close as possible to the actual API so that we don't go to much out of scope of what we are trying here. Perhaps I need to make a function which is called setContent and it sets an array with only a single piece and another one called setContentList() that accepts a list?